react/react1/week3/reactweek3submit/reactsubmit/src/SubmitShift.js

```diff
@@ -0,0 +1,107 @@
+import "./App.css";
+import React, { useState, useEffect } from "react";
+import { ViewShift } from "./ViewShift";
+function SubmitShift() {
+  const [inputShift, setInputShift] = useState([
+    { id: 1, name: "Suman", startTime: "3:20", endTime: "12:33" },
+    { id: 2, name: "Chunmei", startTime: "15:20", endTime: "21:49" },
+  ]);
+  const [name, setName] = useState("");
+  const [startTime, setStartTime] = useState("");
+  const [endTime, setEndTime] = useState("");
+  const [searchTerm, setSearchTerm] = useState("");
+
+  //fetch part not working for me.
+  // useEffect(() => {
+  //   if (setInputShift) {
+  //     fetch(
+  //       "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw"
+  //     )
+  //       .then((res) => res.json())
+  //       .then((data) => {
+  //         const fetchMap = data.map((data) => ({
+  //           name: data.name,
+  //           startTime: data.startTime,
+  //           endTime: data.endTime,
+  //         }));
+  //         setInputShift((prev) => [...prev, ...fetchMap]);
+  //       });
+  //   }
+  // }, [setInputShift]);
```

Comment:
What was the issue with fetching the data?

Reply:
its working